sdk/core/Azure.Core/tests/TestFramework/TestRecording.cs

```diff
@@ -183,16 +183,17 @@ public void Dispose()
         public T InstrumentClientOptions<T>(T clientOptions) where T : ClientOptions
         {
             clientOptions.Transport = CreateTransport(clientOptions.Transport);
+            clientOptions.Retry.Delay = TimeSpan.Zero;
             return clientOptions;
         }
 
         public HttpPipelineTransport CreateTransport(HttpPipelineTransport currentTransport)
         {
             return Mode switch
             {
-                RecordedTestMode.Live => currentTransport,
-                RecordedTestMode.Record => new RecordTransport(_session, currentTransport, entry => !_disableRecording.Value, Random),
-                RecordedTestMode.Playback => new PlaybackTransport(_session, _matcher, Random),
+                RecordedTestMode.Live => (HttpPipelineTransport)new FailureSimulationTransport(currentTransport),
```

Comment:
Is this cast technically needed for something or was it just to match previous behavior?